diffrax/_event.py

```diff
@@ -21,7 +22,32 @@ class Event(eqx.Module):
     """
 
     cond_fn: PyTree[Callable[..., Union[BoolScalarLike, RealScalarLike]]]
-    root_finder: Optional[optx.AbstractRootFinder] = None
+    bidirect: PyTree[bool]
+    root_finder: Optional[optx.AbstractRootFinder]
+
+    def __init__(
+        self,
+        cond_fn,
+        root_finder: Optional[optx.AbstractRootFinder] = None,
+        bidirect: Union[bool, PyTree[bool]] = True,
+    ):
+        vals, treedef = flatten(cond_fn)
+        n = len(vals)
+
+        if isinstance(bidirect, bool):
+            bidir_list = [bidirect] * n
+        else:
+            bidir_list = list(bidirect)
```

Comment:
I think the logic here does not reflect the pytree structure of `cond_fn`. It seems to operate on the flattened representation.

Can we add a test for this case?